From 8f8eed18ed794eede0cb197ff46b121ccd7754d8 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Fri, 7 Jul 2023 09:34:13 +0600
Subject: [PATCH 062/157] drivers: net: lan865x: update thread names

Instead of using constant strings, make thread names dependent on spi
device name. This makes it possible to match instances against devices
in 'ps' output.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>

diff --git a/drivers/net/ethernet/microchip/lan865x/lan865x.c b/drivers/net/ethernet/microchip/lan865x/lan865x.c
index e2eb72522fdf..4ecdc0719a26 100755
--- a/drivers/net/ethernet/microchip/lan865x/lan865x.c
+++ b/drivers/net/ethernet/microchip/lan865x/lan865x.c
@@ -886,14 +886,16 @@ static int lan865x_probe(struct spi_device *spi)
 	timer_setup(&priv->tc6_timer, timer_callback, 0);
 	mod_timer(&priv->tc6_timer, jiffies + msecs_to_jiffies(TIMEOUT));
 
-	priv->irq_task = kthread_run(irq_handler, priv, "irq_task");
+	priv->irq_task = kthread_run(irq_handler, priv, "%s irq_task",
+				     dev_name(&spi->dev));
 	if (IS_ERR(priv->irq_task)) {
 		ret = PTR_ERR(priv->irq_task);
 		goto error_irq_task;
 	}
 	sched_set_fifo(priv->irq_task);
 
-	priv->ns_task = kthread_run(ns_handler, priv, "ns_task");
+	priv->ns_task = kthread_run(ns_handler, priv, "%s ns_task",
+				    dev_name(&spi->dev));
 	if (IS_ERR(priv->ns_task)) {
 		ret = PTR_ERR(priv->ns_task);
 		goto error_ns_task;
-- 
2.25.1

